### PR TITLE
Refactor Escaper

### PR DESCRIPTION
**Pull request by deathaxe**
1. drop `Escaper` base command
2. use pre-compiled patterns

### CODE REVIEW

File: string_encode.py [hunk 1]
```diff
@@ -305,18 +305,15 @@ def encode(self, text):
...
+        return self.regex.sub(r'\\\1', text)
 
 
-class EscapeRegexCommand(Escaper):
-    meta = r'[?\\*.+^$()\[\]\{\}\|]'
-
-
-class EscapeLikeCommand(Escaper):
-    meta = r'[%_]'
+class EscapeLikeCommand(EscapeRegexCommand):
```
Comment (colinta) [inline string_encode.py:315]: I'd prefer a common parent class, but I can make that change. I like the precompiled regex 👍

**Review by colinta — APPROVED** [review]

File: string_encode.py (hunk 1 above)
Comment (deathaxe) [reply string_encode.py:315]: Actually with only two commands an trivial implementation, I'd even just copy `run` method, TBH.

Comment (deathaxe) [reply string_encode.py]: Pushed a change accordingly. I'd inherit from a base class only when more complex implementations or many sub-commands existed.

**Comment by colinta** [comment]
Cool, super simple, love it.

**Pull request merged**